full-node/sov-stf-runner/src/runner.rs

```diff
@@ -201,6 +201,19 @@ where
                     .prove(header_hash)
                     .await
                     .expect("The proof creation should succeed");
+
+                loop {
+                    let status = self.prover_service.send_proof_to_da(header_hash).await;
+                    match status {
+                        crate::ProofSubmissionStatus::Success => {
+                            break;
+                        }
+                        crate::ProofSubmissionStatus::ProvingInProgress => {
+                            tokio::time::sleep(tokio::time::Duration::from_millis(100)).await
```

Comment:
instead of managing the interval, consider delegating this task to [tokio::select](https://docs.rs/tokio/1.34.0/tokio/macro.select.html)

Tokio has a great mechanism to set a fair interval & task pick based on very complicated criteria - we can delegate them this responsibility

Reply:
I would postpone the task for two reasons.:

1.  https://github.com/Sovereign-Labs/sovereign-sdk/pull/1160#discussion_r1399107880 (the `fork-menager` is ready) so we have a better sense of how exactly this method is used
2. #1165 We have a means to test the code better

Reply:
It would however cover a potential edge case where the service is locked with such STF, and an unreasonable timeout is reached. In short, the suggestion would look like

```rust
/// Select will trigger whatever event happens first, returning it
let response = tokio::select! {
  v = self.prover_service.send_proof_to_da(header_hash) => v,
  /// This would halt the entire request if the timeout is reached
  v = tokio::time::sleep(tokio::time::Duration::from_millis(100)) => panic!("timeout"),

};
if let crate::ProofSubmissionStatus::Err(e) = response {
  // report error
}
```